js/core/tty/readline.js

```diff
@@ -0,0 +1,157 @@
+// Copyright 2014-2015 runtime.js project authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+exports.get = function(cb) {
+  var tty = runtime.tty;
+  var inputText = '';
+  var inputPosition = 0;
+  var history = [''];
+  var historyPosition = 0;
+  var done = false;
+  var inputEnabled = true;
+  var endresult = '';
+
+  function drawCursor() {
+    var char = ' ';
+    if (inputPosition < inputText.length) {
+      char = inputText[inputPosition];
+    }
+
+    tty.print(char, 1, tty.color.WHITE, tty.color.LIGHTGREEN);
+    tty.moveOffset(-1);
+  }
+
+  function removeCursor() {
+    var char = ' ';
+    if (inputPosition < inputText.length) {
+      char = inputText[inputPosition];
+    }
+
+    tty.print(char, 1, tty.color.WHITE, tty.color.BLACK);
+    tty.moveOffset(-1);
+  }
```

Comment:
Because it doesn't make any sense to have the same exact methods in multiple places. So, perhaps create a shared module that both `shell/readline.js` and `shell/input-box.js` can use that includes these methods.

I don't undestand how that would be reinventing the wheel?